pkg/process/net/resolver/resolver.go

```diff
@@ -0,0 +1,65 @@
+package resolver
+
+import (
+	"sync"
+	"time"
+
+	model "github.com/DataDog/agent-payload/process"
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+)
+
+const defaultTTL = 10 * time.Second
+
+// LocalResolver is responsible resolving the raddr of connections when they are local containers
+type LocalResolver struct {
+	mux         sync.RWMutex
+	addrToCtrID map[addr]string
+	updated     time.Time
+}
+
+type addr struct {
+	ip    string
+	port  int32
+	proto string
+}
+
+// LoadAddrs generates a map of network addresses to container IDs
+func (l *LocalResolver) LoadAddrs(containers []*containers.Container) {
+	l.mux.Lock()
+	defer l.mux.Unlock()
+
+	if time.Now().Sub(l.updated) < defaultTTL {
+		return
+	}
+
+	l.updated = time.Now()
+	l.addrToCtrID = make(map[addr]string)
+	for _, ctr := range containers {
+		for _, networkAddr := range ctr.AddressList {
+			addr := addr{
+				ip:    networkAddr.IP.String(),
+				port:  int32(networkAddr.Port),
+				proto: networkAddr.Protocol,
+			}
+			l.addrToCtrID[addr] = ctr.ID
+		}
+	}
+}
+
+// Resolve binds container IDs to the Raddr of connections
+func (l *LocalResolver) Resolve(c *model.Connections) {
```

Comment:
That definitely sounds a better approach. I'll update it.

Reply:
Just realized that approach won't work (at least exactly as it is).
basically the PID of the connection that has `hostIP:forwardedPort` on the `Laddr` will be the `PID` of the docker-proxy process

Reply:
We spoke off thread. This would  require a bunch of gnarly changes in dockerproxy, and we can rely on container addresses being present in bridge mode.